lib/semian/circuit_breaker.rb

```diff
@@ -24,6 +25,7 @@ def acquire
       result = nil
       begin
         result = yield
+        mark_failed(result) if result.is_a?(::Net::HTTPServerError) && @open_circuit_server_errors
```

Comment:
This can't leak into the circuit breaker. You must contain this in the `::Net::HTTP` adapter. You can look at MySQL which needs to do something similar to this, I think.